denlillemand/week4/ass5/app/components/namedlist.jsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+import Message from './message.jsx';
+import { swapMessage, createMessage, deleteMessage, archiveMessage } from '../actioncreators/listactions.js';
+
+export default class NamedList extends React.Component {
+    constructor(props, context) {
+        super(props, context);
+        this.state = {
+            messageSwapDestination: "",
+            message: "",
+            sortedMessages: [],
+            messageSwapId: "",
+            messageId: 1
+        };
+        this.onMessageSubmit = this.onMessageSubmit.bind(this);
+        this.onChangeMessageSwapDestinationListName = this.onChangeMessageSwapDestinationListName.bind(this);
+        this.onChangeMessage = this.onChangeMessage.bind(this);
+        this.onChangeMessageSwapId = this.onChangeMessageSwapId.bind(this);
+        this.archiveMessage = this.archiveMessage.bind(this);
+        this.deleteMessage = this.deleteMessage.bind(this);
+        this.submitSwapMessage = this.submitSwapMessage.bind(this);
+    }
+
+    submitSwapMessage(event) {
+        swapMessage(parseInt(this.state.messageSwapId, 10), this.state.messageSwapDestination, this.props.name);
+        this.setState({
+            messageSwapId: "",
+            messageSwapDestination: ""
+        });
+    }
+
+    onChangeMessage(event) {
+        if (event.target.value.length > 200) {
```

Comment:
It can be discussed whether this conflicts with an assignment target

> From now on, no views/components must contain any validation code).

An argument could be that you are validating on component data because the data here is not used for creation until actaully executing the onMessageSubmit function.